common/transport/mqtt/src/test/java/org/thingsboard/server/transport/mqtt/MqttTransportHandlerTest.java

```diff
@@ -205,4 +213,20 @@ public void givenMessageQueue_whenProcessMqttMsgConcurrently_thenEnqueueRegularS
         messages.forEach((msg) -> verify(handler, times(1)).processRegularSessionMsg(ctx, msg));
     }
 
-}
\ No newline at end of file
+    @Test
+    public void givenMqttConnectMessageAndPublishProvisionImmediately_whenProcessMqttMsg(){
+        givenMqttConnectMessage_whenProcessMqttMsg_thenProcessConnect();
```

Comment:
This method will create a regular session connection and if you put then message to handler.channelRead - it will go to the wrong branch of code and will be processed as a regular session message, not like provision message.